### PR TITLE
"add workflow-templates EsLint and audit"

### LINKED ISSUE #3: [C4GT Community]: Configure GitHub Actions to run ESLint on changed files and npm audit on package.json changes
### Description
The [AMRIT UI repos](https://github.com/orgs/PSMRI/repositories?q=-UI), part of the AMRIT (Accessible Medical Records via Integrated Technologies) initiative by Piramal Swasthya, is built with Angular. To maintain high coding standards and ensure the reliability of the application, we need to configure GitHub Actions to automate quality checks before any code is merged into the main branch. This includes enforcing linting, checking for npm vulnerabilities when `package.json` is modified, and ensuring that all tests pass.

#### What You Will Learn:
As a contributor, you will:

- Gain hands-on experience with GitHub Actions and CI/CD pipelines.
- Learn how to automate code quality checks and security scans in Angular projects.
- Develop skills in configuring linting, testing, and dependency vulnerability checks.
-  Work closely with experienced developers to maintain high standards in code quality and project reliability.
- Understand best practices in DevOps and continuous integration.

#### Desired Skills:

- Basic knowledge of Angular and front-end development.
- Familiarity with GitHub, GitHub Actions, and CI/CD concepts.
- Understanding of linting tools (e.g., ESLint) and package managers (e.g., npm).
- Eagerness to learn about DevOps practices and automation tools.

### Goals

- [ ] Set up GitHub Actions - create a GitHub Actions workflow for the AMRIT UI repository; configure the workflow to run linting checks using ESLint on every pull request.
- [ ] Linting Enforcement - ensure that code cannot be merged unless linting passes without errors.
- [ ] Vulnerability Scanning - set up npm audit in the GitHub Actions workflow to check for vulnerabilities when package.json or package-lock.json files are changed; ensure that critical vulnerabilities must be resolved before merging.

### Expected Outcome

A fully configured GitHub Actions pipeline that enforces coding standards, reliability checks, and security scans for the AMRIT UI repositories. This setup will ensure that only clean, secure, and well-tested code is merged, maintaining the integrity of the application. The workflows must be configured for all the UI repos.
[https://github.com/orgs/PSMRI/repositories?q=-UI](https://github.com/orgs/PSMRI/repositories?q=-UI)

### Acceptance Criteria

- GitHub Actions workflow is set up and running for every pull request.
- Linting errors are detected and must be resolved before merging.
- npm audit checks for vulnerabilities when dependencies are modified.

### Implementation Details

https://github.com/typicode/husky
https://github.com/marketplace/actions/run-eslint
https://github.com/marketplace/actions/npm-audit-action

### Mockups/Wireframes

NA

### Product Name

AMRIT

### Organisation Name

Piramal Swasthya Management Research Institute

### Domain

Healthcare

### Tech Skills Needed

Angular, ESLint, JavaScript, TypeScript

### Mentor(s)

@drtechie

### Complexity

Medium

### Category

CI/CD, Deployment, Testing

### PR DESCRIPTION
**Pull request by PayalKumari10**
### Related Issue
**Fixes [#3](https://github.com/PSMRI/AMRIT/issues/3)**

### Description
This pull request configures GitHub Actions to improve code quality and security monitoring. It includes:
- **ESLint Workflow**: Runs ESLint on code changes in the `main` branch to ensure code quality.
- **npm Audit Workflow**: Runs `npm audit` on changes to `package.json` or `package-lock.json`  to check for vulnerabilities.

### Type of PR
- **Feature enhancement**

### Acceptance Criteria

- GitHub Actions workflow is set up and running for every pull request.
- Linting errors are detected and must be resolved before merging.
- npm audit checks for vulnerabilities when dependencies are modified.

### Additional Notes
- The workflow templates are added in the `.github/workflow-templates/` directory and referenced via `workflow_call` in other repositories.
- Review the workflow files for accuracy and test the workflows after merging.

- **New Features**
	- Introduced an ESLint template for automated linting checks on JavaScript and TypeScript files.
	- Added a GitHub Actions workflow for continuous linting on code pushes and pull requests.
	- Implemented an NPM Audit template to check for vulnerabilities in project dependencies.
	- Created a GitHub Actions workflow for security audits triggered by changes to dependency files.

- **Impact**
	- Enhances code quality and security by automating linting and vulnerability checks during development.

### CODE REVIEW

File: workflow-templates/eslint.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: Linting
+
+on:
+  push:
+    branches:
+      - main
```
Comment (drtechie) [inline workflow-templates/eslint.yml:6]: Let's add main, master and develop branches

File: workflow-templates/eslint.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: '16'
+      - run: npm install
+      - run: npm run lint
```
Comment (drtechie) [inline workflow-templates/eslint.yml]: So we need to only lint changed files in the PR

File: workflow-templates/npm-audit.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+jobs:
+  audit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install dependencies
+        run: npm install
+      - name: Run npm audit
+        run: npm audit
+        continue-on-error: false  # Ensure PRs are blocked if vulnerabilities are found
```
Comment (drtechie) [inline workflow-templates/npm-audit.yml]: We can skip this for now.
Sometimes there won't be fixes yet for new vulnerabilities

File: workflow-templates/eslint.yml (hunk 1 above)
Comment (PayalKumari10) [reply workflow-templates/eslint.yml:6]: okay i will add

File: workflow-templates/eslint.yml (hunk 2 above)
Comment (PayalKumari10) [reply workflow-templates/eslint.yml]: sure

File: workflow-templates/npm-audit.yml (hunk 3 above)
Comment (PayalKumari10) [reply workflow-templates/npm-audit.yml]: Ok  @drtechie  sir, I have done it, please have a look.

**Review by drtechie — APPROVED** [review]

**Comment by VedantKhairnar** [comment]
Hey there,
Thanks for your contribution.
As part of Augtoberfest, we want to reward you for your contributions, so pls fill out [this form](https://forms.gle/VBDUPWMYcZEbmVrQ9) by 19th Sept.
Thanks.

**Comment by drtechie** [comment]
@PayalKumari10 will you create the follow-up PRs for implementing the Actions in UI repos?
Please provide an ETA.

**Comment by PayalKumari10** [comment]
> @PayalKumari10 will you create the follow-up PRs for implementing the Actions in UI repos? Please provide an ETA.

Ok @drtechie sir , As soon as I complete this work, I will inform you. As soon as possible. Thanks

**Pull request merged**